### PR TITLE
[SCSB-145] require Python 3.10

### PR DESCRIPTION
**Pull request by zacharyburnett**
resolves [SCSB-145](https://jira.stsci.edu/browse/SCSB-145)

propagate Astropy's deprecation of Python 3.9 to downstream packages

> [!NOTE]
> this PR was generated automatically by ``batchpr`` :robot:

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -3,61 +3,74 @@ name = "jwql"
...
+    "scipy",
+    "sqlalchemy",
+    "stdatamodels",
+    "wtforms",
+]
+dynamic = [
+    "version",
 ]
-dynamic = ["version"]
+requires-python = "3.10"
```
Comment (zacharyburnett) [inline pyproject.toml]:
```suggestion
requires-python = ">=3.10"
```

**Comment by zacharyburnett** [comment]
The `Required Status Checks` at https://github.com/spacetelescope/jwql/settings/branch_protection_rules will have to be updated to remove the following:
```
test (Python 3.9, macos-latest)
test (Python 3.9, ubuntu-latest)
```

**Review by mfixstsci — APPROVED** [review]
Talk in #1546 brought me back to this. Sorry for the long wait @zacharyburnett this has been on my to-do list for a while now.

**Comment by mfixstsci** [comment]
Merging with failing test, this is expected because `redis` now splits server and python modules into `redis-py` and `redis-server` as dependency names. Fix is in #1546

**Pull request merged**